patches/generate-xform.patch

```diff
@@ -3,7 +3,7 @@
 *** 6,13 ****
   const path = require('path');
   const htmlParser = require('node-html-parser');
-  const logger = require('../logger');
+  const logger = require('@medic/logger');
```

Comment:
@jkuester I'm have a sticking point as I add CHT-Core 4.8.2 into the harness and requesting your thoughts. Similarly, how could we support a feature branch which makes a significant change to harness functionality?

This patch file needs to change to successfully apply; the same file won't work on 4.6 and 4.8, so we would need to keep two versions. Currently, you rebuild all previous versions of cht-core during `build-ci` (via --force) but I'm wondering about whether that decision is the right one.  All the build outputs are committed to the repository; so I'm doubting the value and effort required to regenerate these. Seems we should (also?) be testing the build artifacts which are in the repo since that is what users would receive (?)

I'm wondering your thoughts and how your instincts would guide you?

In this PR, I'm building off a feature branch so things get kinda weird. I remove `--force` form `npm run build-ci`, and I remove 4.6 from build.sh. I'm not exactly proposing that; but it's just the lens I'm coming to this conversation with.

Reply:
> Currently, you rebuild all previous versions of cht-core during build-ci (via --force) but I'm wondering about whether that decision is the right one.

It almost certainly is not the right one!  :sweat_smile: I knew when I did that this day would come. The end-goal of having the `dist/cht-core-*` directories be independent artifacts was to make our lives easier now.  The only reason I included the `--force` command (and triggered it on CI) was because it worked at the time and I knew that by doing that I could kick this question down the road....  :facepalm: 

Here are the tradeoffs that I see at play here: 

If we treat the  `dist/cht-core-*` artifacts as _permanent_ and do not support re-building them as needed, then we are going to prevent some new test harness features (specifically ones that require additional imports from cht-core) from being compatible with older cht-core targets.  One example is https://github.com/medic/cht-conf-test-harness/issues/83 where we may decide the harness needs to depend on some logic from cht-core for parsing the form expressions. And any of the cht-core targets that we cannot rebuild to include the additional code will not be able to support the new functionality.  Then it seems like we will have to drop support for that cht-core target from new versions of the test-harness.

On the other hand, it seems unlikely that we will be able to support continuing to build old cht-core versions in perpetuity. The older things get, the more cursed the build environment will need to be to try and deal with it.  It could easily turn into a huge waste of time trying to manage this and ultimately slow down test-harness development.  

---

My current thinking on the best approach here is:
- We should continue re-building as many of the previous CHT versions as possible automatically on our CI. (e.g. controlled by the contents of the `cht_versions` array in build.sh)
- When we add support to the test-harness for a new version of cht-core and that requires a breaking change to our build process (such as what we have now), we stop building the broken CHT versions (remove them from `cht_versions`), but keep their `dist/cht-core-*` artifacts.
    - We can still ship this change as a new _minor_ version for the test harness since all the harness functionality should still work with all the `dist/cht-core-*` versions.
    - As you noted, it would be important to have the automated tests run against each "supported" cht-core version to make sure functionality is maintained.
- When we add new test-harness features that do not require additional code from cht-core (e.g. fixing bugs in the test-harness code), these can be released as minor/patch versions of the test-harness and should be compatible with all currently supported cht-core versions.
- When we add a new test-harness feature that _requires_ additional code from cht-core such that the `dist/cht-core-*` artifacts need to be rebuilt, then we only support that feature for whatever versions of the CHT that are currently being auto-built on CI.  
  - This means that we will have to drop support from the test-harness for any cht-core versions that are not currently being built, which IMHO should be done in a new _major_ version of the test-harness.
  - Unsupported CHT versions should be removed from `dist/cht-core-*`.

To put things more simply, I think we need to recognize two (potentially different) sets of CHT versions: 
- The CHT version targets currently supported by the test harness - these are the ones contained in `dist/cht-core-*`.
- The CHT versions supported by our build script - these are the ones contained in he `cht_versions` array in build.sh.

Changes to the `cht_versions` array should not be considered breaking changes, but changes to `dist/cht-core-*` are breaking changes.

My hope with this approach is that it allows us to easily add support for new cht-core versions while maintaining support for as many previous cht-core versions as possible.

Reply:
Nice. I concur. Got some time to review this latest iteration?